docs/DesignDocs/system_ocall_opt_in.md

```diff
@@ -0,0 +1,120 @@
+# Explicit Enclave Opt-in to system ocalls
+
+System ocalls are ocalls that are part of openenclave core libraries. These are generally not
+called by the enclave developer but instead are called by other functions internal to OE SDK.
+
+Because ocalls have potential security impact, and enclave developer should be able to opt
```

Comment:
We should elaborate on the motivation for this proposal; referring to vague "potential security impact" tends to generate FUD.

There are two major concerns here:
* A developer needs to be aware of what information or functionality an enclave is reaching out to the untrusted host for, e.g. time syscalls, so that they can be evaluated in the app's security model. Having this be an explicit developer opt-in makes it much more obvious to a developer.
* A developer needs to be aware of when the enclave is calling out to the untrusted host. For example, they need to know if calling an OE function will result in an OCall, and understand that they must not hold enclave locks across such calls. The opt-in mechanism introducing link failures can help surface awareness of OE calls being used that rely on OCalls that would not otherwise be noticed.

Reply:
@jhand2 I don't believe this was addressed although a force push has unanchored this comment.

Reply:
Updated the doc to be more clear about the concern:

> Because information from insdie the enclave may be transmitted to the untrusted host
during an OCalls, an enclave developer should be able to opt-out of these OCalls
being made. They should also have visibility into what OCalls might be called by OE SDK.